CMakeLists.txt

```diff
@@ -21,6 +21,76 @@ if (NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
   message(STATUS "Setting unspecified CMAKE_BUILD_TYPE to '${CMAKE_BUILD_TYPE}'. This is the correct setting for normal builds.")
 endif()
 
+set(IRODS_BUILD_DOCS_DEFAULT ON)
+
+# Find help2man
+if (NOT DEFINED BUILD_DOCS OR BUILD_DOCS)
+  if (NOT DEFINED HELP2MAN)
+    set(HELP2MAN_SET_BY_CMAKE ON)
+  endif()
+  find_program(
+    HELP2MAN
+    NAMES help2man
+    DOC "help2man location"
+    )
+  if (DEFINED HELP2MAN)
+    message(STATUS "Found help2man: ${HELP2MAN}")
+    if (HELP2MAN_SET_BY_CMAKE)
+      # check version number, but only if find_program went fishing
+      execute_process(
+        COMMAND "${HELP2MAN}" --version
+        OUTPUT_VARIABLE HELP2MAN_VERSION
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+      string(REGEX REPLACE "\n.*" "" HELP2MAN_VERSION "${HELP2MAN_VERSION}")
+      string(REGEX REPLACE "[^ ]+ +[^ ]+ +" "" HELP2MAN_VERSION "${HELP2MAN_VERSION}")
+      message(STATUS "Found help2man version ${HELP2MAN_VERSION}")
+      if (HELP2MAN_VERSION VERSION_LESS "1.37")
+        # don't cache unsable help2man
+        unset(HELP2MAN CACHE)
+        if (NOT DEFINED BUILD_DOCS)
+          set(IRODS_BUILD_DOCS_DEFAULT OFF)
+        else()
+          message(SEND_ERROR "help2man version 1.37 or later required, cannot build docs")
+        endif()
+      endif()
+    endif()
+  else()
+    if (NOT DEFINED BUILD_DOCS)
+      set(IRODS_BUILD_DOCS_DEFAULT OFF)
+    else()
+      message(SEND_ERROR "help2man not found, cannot build docs")
+    endif()
+  endif()
+endif()
+
+# Find gzip
+if (NOT DEFINED BUILD_DOCS OR BUILD_DOCS)
+  if (NOT DEFINED GZIP)
+    set(GZIP_SET_BY_CMAKE ON)
+  endif()
+  find_program(
+    GZIP
+    NAMES gzip pigz
```

Comment:
i didn't know about pigz!

Reply:
I wanted to stick `bgzip` in there, too, but it doesn't support all the `gzip` arguments we use.